examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions include a small-ampitude perturbation that decays away from the center
+# of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2)
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 + ϵ * exp(- (y - Ly/2)^2 ) * randn()
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2)
+ hⁱ(x, y, z) = model.grid.Lz + ηⁱ(x, y, z)
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z)
+nothing # hide
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
+   ω_pert = ComputedField( ω_field - Ω )
+
+# ## Running a `Simulation`
+#
+# We pick the time-step that ensures to resolve the surface gravity waves.
+# A time-step wizard can be applied to use an adaptive time step.
+
+simulation = Simulation(model, Δt = 1e-2, stop_time = 150.00)
+
+# ## Prepare output files
+#
+# Define a function to compute the norm of the perturbation on the cross channel velocity.
+# We obtain the `norm` function from `LinearAlgebra`.
+
+using LinearAlgebra, Printf
+
+function perturbation_norm(model)
+    compute!(v)
+    return norm(interior(v))
+end
+nothing # hide
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (perturbation_norm = perturbation_norm,),
+        filepath=joinpath(@__DIR__, "perturbation_norm_shallow_water.nc"),
+        schedule=IterationInterval(1),
+        dimensions=(perturbation_norm=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visualize the results
+#
+
+using NCDatasets
+nothing # hide
+
+# This sets the environment to avoid an error in GKSwstype
+
+ENV["GKSwstype"] = "100"
+
```

Comment:
```suggestion
```

Reply:
I don't understand why this appears here. I'm pretty sure it's not needed (except if the users is running Oceananigans on a remote machine via `ssh -Y ...`).

Reply:
I found that without this line the `@animate` look froze.  I don't pretend to know why but @ali-ramadhan confirmed the weiird behaviour and we found this worked.  I admit that it looks odd, and I would also prefer it not to be there, but I think we need to keep it here, until `Plots.jl` fixed the bug.  Sorry.

Reply:
Unfortunately, @ali-ramadhan and I found that if we didn't include this `@animate` would freeze and never finish.  I think we need this until `Plots.jl` fixes the bug.  Sorry.

Reply:
What Plots.jl bug?
The environment variable is set by `docs/make.jl`; no need to add it inside the example and have users wonder about what this is...

Reply:
I'm removing it now to see if docs build ok. I believe they will.
(This may be just me banging my head on the wall... let's see...)